src/index.d.ts

```diff
@@ -38,10 +38,9 @@ export interface Color {
  * Utilies types based on:
  * https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-307871458
  */
-export type Diff<T extends string, U extends string> = ({ [P in T]: P } &
+type Diff<T extends string, U extends string> = ({ [P in T]: P } &
   { [P in U]: never } & { [x: string]: never })[T];
-export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
-export type Replace<T, S> = Omit<T, keyof S & keyof T> & S
+type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
```

Comment:
`Replace` is no longer needed, I think, and `Diff` and `Omit` don't need to be exported since they're only used in this file.

Reply:
Sure, but this way people can't use these types in their own projects anymore, if they want :)

Reply:
Sure they can, they can copy these definitions into their own library, just as mui did, or else use a library like `fp-ts` that provides them. Does it really make sense for mui to provide these things? What do they have to do with material design?

Reply:
More to the point, my understanding was that these were never meant to be part of the public "API" of `material-ui`. They were only exported here because many different components needed to use them in their typings, whereas now all uses of `Omit` have been abstracted over via `StandardProps`. Even `StandardProps` is still not part of the public API of `material-ui` and we should feel free to replace it or get rid of it in the future as need be. Perhaps it'd be a good idea to add `/** @internal */` comments to such exports to clarify that.

Reply:
But where is the harm in exposing them as part of the API? If we need to change them in the future, we'll let the users know that there is a *BREACKING CHANGE*.

IMHO these helper types come in handy, when you add your own abstraction onto `material-ui`. For example, if you want to make some property fix. I would rather not have to copy `Omit` and `Diff` all the time to new projects.

Reply:
I'm not sure Material-UI is the best location to host those helpers. If we want to make them public, it would be much better to document them.

Reply:
@sebald the harm is that they have nothing to do with material design nor are they typically needed to make use of the library, so they’re just cluttering up the API surface. AFAIK none of the documentation refers to them because they’re simply not needed by downstream users. But if you do find you need them for whatever reason the appropriate way to get them is by depending on a library like `fp-ts`.

Reply:
Ok, guess I am over-voted 😀